### PR TITLE
Updates Ansible for Install Doc Fixes

### PR DESCRIPTION
**Pull request by danehans**
Previously, the installation guide and gemini ansible code
was out of sync. This commit provides several updates to sync
the two.

### CODE REVIEW

**Comment by danehans** [comment]
Supports the doc updates in https://github.com/gemini-project/gemini/pull/26

File: provisioners/ansible/pre-setup.sh [hunk 1]
```diff
@@ -1,17 +1,25 @@
 #!/bin/bash
 
-base_dir=${BASE_DIR:-~/gemini/provisioners/ansible/playbooks/roles}
+base_dir=${BASE_DIR:-~/gemini/provisioners/ansible/roles}
+contrib_base_dir=${CONTRIB_BASE_DIR:-~}
+
+echo "-> Setting Up SSH"
+ssh-keygen -t rsa -N "" -f ~/.ssh/id_rsa
+cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
```
Comment (stephenrlouie) [inline provisioners/ansible/pre-setup.sh]: Potential ssh_key to the group_vars/all.yml

File: provisioners/ansible/pre-setup.sh [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+echo "-> Cloning Repos"
+git clone https://github.com/gemini-project/contrib.git ${contrib_base_dir}/contrib
 git clone https://github.com/gemini-project/ansible-role-ansible-controller.git ${base_dir}/ansible-controller
 git clone -b pxe_coreos https://github.com/gemini-project/ansible-role-httpd.git ${base_dir}/httpd
 git clone -b pxe_coreos https://github.com/gemini-project/ansible-role-tftp.git ${base_dir}/tftp
 git clone -b pxe_coreos https://github.com/gemini-project/ansible-dnsmasq.git ${base_dir}/dnsmasq
 git clone -b pxe_coreos https://github.com/gemini-project/ansible-coreos-cloudinit.git ${base_dir}/coreos-cloudinit
 
 # Show the populated roles directory
+echo "-> These are your Ansible roles"
```
Comment (stephenrlouie) [inline provisioners/ansible/pre-setup.sh]: Possibly a better description of what this is. gemini_master/roles and contrib/roles

File: provisioners/ansible/gemini-master [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[gem-master]
+gem-master01 ansible_host=127.0.0.1
```
Comment (stephenrlouie) [inline provisioners/ansible/gemini-master]: Hostname? I usually see IP only.

File: provisioners/ansible/group_vars/all.yml [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
+    mac: 'AA:BB:44:33:22:11'
+    ip: '10.10.10.10'
+# repeat:  - name:/mac:/ip: for additional hosts
+
+# Gateway IP for PXE-booted nodes
+dnsmasq_option_router: '10.10.10.1'
+
+# Replace ssh-rsa ${RSA_KEY} with the RSA key of your gemini-master node.
+cloudinit_ssh_authorized_keys: >
+  ${RSA_KEY}
```
Comment (stephenrlouie) [inline provisioners/ansible/group_vars/all.yml]: Ansible fact? some automation for RSA_KEY since this will trip people up.

**Comment by stephenrlouie** [comment]
LGTM. Big improvement on what is existing, well done!

**Comment by danehans** [comment]
I'm going to merge this and we can fix any minor bugs and/or improvements in a future PR.

**Pull request merged**